examples/pytorch/ogb/ngnn/main.py

```diff
@@ -0,0 +1,437 @@
+    parser.add_argument('--hidden_channels', type=int, default=256)
+    parser.add_argument('--dropout', type=float, default=0.0)
+    parser.add_argument('--batch_size', type=int, default=64 * 1024)
+    parser.add_argument('--lr', type=float, default=0.001)
+    parser.add_argument('--epochs', type=int, default=400)
+
+    # training settings
+    parser.add_argument('--use_valedges_as_input', action='store_true', help='Use training + validation edges for inference on test set.')
+    parser.add_argument('--log_steps', type=int, default=1)
+    parser.add_argument('--eval_steps', type=int, default=1)
+    parser.add_argument('--runs', type=int, default=10)
+    global args; args = parser.parse_args()
+    print(args)
+
+    device = f'cuda:{args.device}' if args.device != -1 and torch.cuda.is_available() else 'cpu'
+    device = torch.device(device)
+
+    dataset = DglLinkPropPredDataset(name=args.dataset)
+    data = dataset[0]
+    split_edge = dataset.get_edge_split()
+
+    # We randomly pick some training samples that we want to evaluate on:
+    idx = torch.randperm(split_edge['train']['edge'].size(0))
+    idx = idx[:split_edge['valid']['edge'].size(0)]
+    split_edge['eval_train'] = {'edge': split_edge['train']['edge'][idx]}
+
+    if args.dataset == 'ogbl-ppa':
+        data.ndata['feat'] = data.ndata['feat'].to(torch.float)
+
+    if args.dataset == 'ogbl-ddi':
+        emb = torch.nn.Embedding(data.num_nodes(), args.hidden_channels).to(device)
+        in_channels = args.hidden_channels
+    else: # ogbl-collab, ogbl-ppa
+        in_channels = data.ndata['feat'].size(-1)
+    
+    # select model
+    if args.use_sage:
+        model = SAGE(in_channels, args.hidden_channels,
+                     args.hidden_channels, args.num_layers,
+                     args.dropout, args.ngnn_type)
+    else: # GCN
+        data = dgl.add_self_loop(data)
+        model = GCN(in_channels, args.hidden_channels,
+                    args.hidden_channels, args.num_layers,
+                    args.dropout, args.ngnn_type)
+
+    valdata = data
+    if args.use_valedges_as_input: # Use training + validation edges for inference on test set.
+        val_edges = split_edge['valid']['edge'].T
+        valdata = dgl.add_edges(data, val_edges[0], val_edges[1], {key: split_edge['valid'][key].view(-1, 1) for key in ['weight', 'year']})
+        valdata = dgl.add_edges(valdata, val_edges[1], val_edges[0], {key: split_edge['valid'][key].view(-1, 1) for key in ['weight', 'year']})
+
+    predictor = LinkPredictor(args.hidden_channels, args.hidden_channels, 1, 3, args.dropout)
+
+    data, valdata, model, predictor = map(lambda x: x.to(device), (data, valdata, model, predictor))
+
+    evaluator = Evaluator(name=args.dataset)
+    loggers = {
+        'Hits@20': Logger(args.runs, args),
+        'Hits@50': Logger(args.runs, args),
+        'Hits@100': Logger(args.runs, args),
+    }
+
+    for run in range(args.runs):
+        torch.cuda.empty_cache()
```

Comment:
Do we have an issue without `torch.cuda.empty_cache()`?

Reply:
No, it's just a habit of mine. I will remove it in the next commit.